subsys/cpu_freq/metrics/cpu_load.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2025 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(cpu_freq_metrics_cpu_load, CONFIG_CPU_FREQ_LOG_LEVEL);
+
+static uint64_t execution_cycles_prev;
+static uint64_t total_cycles_prev;
+
+int cpu_freq_metric_get_cpu_load(uint32_t *load)
```

Comment:
This should be split out and provided on its own, does not need to be part of this subsystem, this way it can be used by others. Same suggestion was made here: https://github.com/zephyrproject-rtos/zephyr/pull/93461#discussion_r2226899062

I just tried moving this out and it works nicely on its own and without dependency on cpu_freq subsystem. We do not want yet another cpu_load utility hidden under a subsystem. The one we have in subsys/debug probably should be folded into this as this is multi-arch and will support multi-core eventually.

Maybe move this to lib/os or kernel/

Reply:
also, naming is inconsistent, you have for example `cpu_freq_pstate_set`, but here you have `cpu_freq_metric_get_cpu_load`,  you should move the `get` to the end.